bigquery/google/cloud/bigquery/client.py

```diff
@@ -353,23 +354,43 @@ def update_dataset(self, dataset, fields, retry=DEFAULT_RETRY):
             retry, method='PATCH', path=path, data=partial, headers=headers)
         return Dataset.from_api_repr(api_response)
 
-    def update_table(self, table, properties, retry=DEFAULT_RETRY):
-        """API call:  update table properties via a PUT request
+    def update_table(
+            self, table, fields=None, properties=None, retry=DEFAULT_RETRY):
```

Comment:
This is already a breaking change because you're

1. Changing `properties` from a positional to a keyword argument
1. Changing `properties` in the "positional order" from 2nd to 3rd (the term "positional order" is not one that is used, I'm just using it as shorthand)

Reply:
Acknowledged, moot since I just renamed the field in my latest commit.

It's a breaking change for a different reason than the ones you provide, though.

(1 & 2).  Actually, the `fields` argument can still be used as a positional parameter, identically how `properties` argument was before. It remains a a [positional-or-keyword parameter](https://docs.python.org/2/glossary.html#term-parameter) after adding a default value. If they used the `properties` argument positionally it works the same and if they used it as a keyword argument it works the same.

(3). The actual breaking change here is with the retry argument. Someone could have used the `retry` argument positionally, which now would get passed to `properties` instead. If we kept with the strategy of deprecating `properties`, I should have added it as a keyword argument to the end or have used `**kwargs` to fetch it.

Reply:
I realize you **can** use a positional as a keyword and pass a keyword as positional, but that opens the door for more states than one could possibly hope to wrangle.

The most relevant surface is what we **provide**, i.e. what is defined in our code and indicated in our docs. I have seen people who consider all arguments to be allowed as positional and this makes updating APIs very difficult.

@jonparrott and I have discussed before and determined that the order of keywords should not be a deciding factor in API changes.